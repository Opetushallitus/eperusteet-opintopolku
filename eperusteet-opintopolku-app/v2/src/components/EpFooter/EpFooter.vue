<template>
  <footer>
    <div class="horizontal-line">
      <a :href="$kaanna(linkit.oph)">
        <img :src="ophLogo" :alt="$t('oph')" />
      </a>
    </div>

    <div class="footer-content">
      <div class="row justify-content-around footer-row">
        <div class="col-lg-2 col-md-6 col-sm-12 mb-5">
          <a :href="$kaanna(linkit.opintopolku)">
            <img class="opintopolku" :src="opintopolkuLogo" :alt="$t('opintopolku')" />
          </a>
        </div>
        <div class="col-lg-2 col-md-6 col-sm-12 mb-5">
          <span class="font-weight-bold">{{ $t('eperusteet-virkailija-linkki') }}</span>
          <ep-linkki :url="$kaanna(linkit.virkailija)" icon="ulkoinen-linkki"></ep-linkki>
        </div>
        <div class="col-lg-2 col-md-6 col-sm-12 mb-5">
          <span class="font-weight-bold">{{ $t('anna-palautetta') }}</span>
          <ep-linkki :url="$kaanna(linkit.palaute)" icon="envelope"></ep-linkki>
        </div>
        <div class="col-lg-2 col-md-6 col-sm-12 mb-5">
          <a :href="$kaanna(linkit.saavutettavuusseloste)">{{ $t('saavutettavuusseloste') }}</a>
        </div>
      </div>
    </div>
  </footer>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import EpLinkki from '@shared/components/EpLinkki/EpLinkki.vue';
import ophLogo from '@assets/img/banners/oph_logo.svg';
import opintopolkuLogo from '@assets/img/banners/opintopolku_logo.svg';

@Component({
  name: 'EpFooter',
  components: {
    EpLinkki,
  },
})
export default class EpFooter extends Vue {
  get ophLogo() {
    return ophLogo;
  }

  get opintopolkuLogo() {
    return opintopolkuLogo;
  }

  get linkit() {
    return {
      oph: {
        fi: 'https://www.oph.fi/fi',
        sv: 'https://www.oph.fi/sv',
        en: 'https://www.oph.fi/en',
      },
      opintopolku: {
        fi: 'https://opintopolku.fi/konfo/fi',
        sv: 'https://opintopolku.fi/konfo/sv',
        en: 'https://opintopolku.fi/konfo/en',
      },
      eperusteet_vanha: {
        fi: 'https://eperusteet.opintopolku.fi',
        sv: 'https://eperusteet.opintopolku.fi',
      },
      virkailija: {
        fi: 'http://virkailija.opintopolku.fi',
        sv: 'http://virkailija.opintopolku.fi',
      },
      palaute: {
        fi: 'mailto:eperusteet@opintopolku.fi',
        sv: 'mailto:eperusteet@opintopolku.fi',
      },
      saavutettavuusseloste: {
        fi: 'https://opintopolku.fi/konfo/fi/sivu/eperusteet-saavutettavuusseloste',
        sv: 'https://opintopolku.fi/konfo/sv/sivu/tillganglighetsutlatande-for-egrunder',
      },
      tietosuojaseloste: {
        fi: 'https://opintopolku.fi/wp/tietosuojaseloste',
        sv: 'https://studieinfo.fi/wp/dataskyddsbeskrivning',
      },
    };
  }
}
</script>

<style scoped lang="scss">
@import '@shared/styles/_variables.scss';

footer {
  padding: 0;
  hyphens: none;
  margin-top: 100px;

  .horizontal-line {
    width: 100%;
    border: none;
    padding: 0;
    overflow: visible;
    text-align: center;
    background-color: #FFF;
    border-top-color: $gray-lighten-3;
    border-top-style: solid;
    border-top-width: 1px;

    img {
      top: -30px;
      padding: 0 50px;
      position: relative;
      background-color: #FFF;

    }
  }

  img {
    width: 300px;

    &.opintopolku {
      width: 200px;
    }
  }

  .footer-content {
    max-width: 90%;
    margin: 0 auto;
    padding: 20px 0;
  }

  @media (max-width: 991.98px) {
    .footer-row {
      display: grid;
    }
  }
}

</style>
