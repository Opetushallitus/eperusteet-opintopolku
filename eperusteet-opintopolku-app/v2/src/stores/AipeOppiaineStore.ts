import Vue from 'vue';
import VueCompositionApi, { reactive, computed, ref, watch } from '@vue/composition-api';
import _ from 'lodash';
import { Aipeopetuksensisalto, AIPEOppiaineDto, AIPEOppiaineSuppeaDto, LaajaalainenOsaaminenDto } from '@shared/api/eperusteet';

Vue.use(VueCompositionApi);

export class AipeOppiaineStore {
  private state = reactive({
    oppiaine: null as AIPEOppiaineDto | null,
    oppimaarat: null as AIPEOppiaineSuppeaDto[] | null,
    laajaAlaisetOsaamiset: null as LaajaalainenOsaaminenDto[] | null,
  })

  public static async create(perusteId: number, vaiheId: number, oppiaineId: number) {
    const result = new AipeOppiaineStore(perusteId, vaiheId, oppiaineId);
    await result.init();
    return result;
  }

  constructor(private perusteId: number, private vaiheId: number, private oppiaineId: number) {
  }

  public readonly oppiaine = computed(() => this.state.oppiaine);
  public readonly oppimaarat = computed(() => this.state.oppimaarat);
  public readonly laajaAlaisetOsaamiset = computed(() => this.state.laajaAlaisetOsaamiset);

  public async init() {
    this.state.oppiaine = (await Aipeopetuksensisalto.getAipeOppiaine(this.perusteId, this.vaiheId, this.oppiaineId)).data;
    this.state.oppimaarat = (await Aipeopetuksensisalto.getAipeOppimaarat(this.perusteId, this.vaiheId, this.oppiaineId)).data;
    this.state.laajaAlaisetOsaamiset = (await Aipeopetuksensisalto.getAipeOsaamiset(this.perusteId)).data;
  }
}
