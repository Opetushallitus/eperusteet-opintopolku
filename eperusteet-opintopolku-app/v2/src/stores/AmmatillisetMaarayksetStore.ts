import Vue from 'vue';
import VueCompositionApi, { reactive, computed } from '@vue/composition-api';
import { MaaraysDto, MuutMaaraykset } from '@shared/api/eperusteet';
import _ from 'lodash';

Vue.use(VueCompositionApi);

export class AmmatillisetMaarayksetStore {
  public state = reactive({
    maaraykset: null as MaaraysDto[] | null,
  })

  constructor() {
    this.fetch();
  }

  public readonly maaraykset = computed(() => this.state.maaraykset);

  public async fetch() {
    this.state.maaraykset = (await MuutMaaraykset.getMuutMaaraykset()).data;
  }
}
