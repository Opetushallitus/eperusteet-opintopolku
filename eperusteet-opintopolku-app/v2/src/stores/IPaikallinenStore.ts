import { Computed } from '@shared/utils/interfaces';
import { OpetussuunnitelmaJulkinenDto } from '@shared/api/ylops';

export interface IPaikallinenStore {
  perusteId?: Computed<number>;
  opetussuunnitelmat: Computed<any[]>;
  opetussuunnitelmatPaged?: Computed<any>;
  fetch: (id?: number, diaarinumero?: string) => Promise<void>
  fetchQuery?: (query: any) => Promise<void>;
};
