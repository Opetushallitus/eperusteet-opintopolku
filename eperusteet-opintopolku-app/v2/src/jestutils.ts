import { PerusteStore } from '@/stores/PerusteStore';
import { PerusteKoosteStore } from '@/stores/PerusteKoosteStore';
import { TiedoteStore } from '@/stores/TiedoteStore';
import { PerusteDataStore } from '@/stores/PerusteDataStore';
import Vue from 'vue';
import _ from 'lodash';
import { RouterLinkStub, Wrapper } from '@vue/test-utils';

import '@/config/bootstrap';
import '@/config/fontawesome';


export function findAllContaining<T extends Vue>(wrapper: Wrapper<T>, selector: string, text: string) {
  return wrapper.findAll(selector).filter(r => r.text().includes(text));
}

export function findContaining<T extends Vue>(wrapper: Wrapper<T>, selector: string, text: string) {
  const results = findAllContaining(wrapper, selector, text);
  if (results.length !== 1) {
    throw new Error('Multiple results: ' + selector + ' ' + text);
  }
  return results.at(0);
}

export const mocks = Object.freeze({
  $t: x => x,
  $kaanna: x => x ? x.fi : 'kaanna',
  $sd: x => x,
  $ld: x => x,
}) as any;

export const stubs = Object.freeze({
  RouterLink: RouterLinkStub,
}) as any;

function wrap<T extends object>(original: T, value: T) {
  const result: any = {};

  // Get original implementations
  for (const k in original) {
    if (_.isFunction(original[k])) {
      result[k] = jest.fn(original[k] as any);
    }
    else {
      result[k] = original[k];
    }
  }

  // Overwrite with default mocks
  _.forEach(value, (v, k) => {
    if (_.isFunction(v)) {
      result[k] = jest.fn(v);
    }
    else {
      result[k] = Vue.observable(v);
    }
  });

  const Mock = jest.fn(() => Vue.observable(result) as T);
  return new Mock();
}

export const perusteKoosteStoreMock = (config: Partial<PerusteKoosteStore> = {}) => {
  return wrap(PerusteKoosteStore.prototype, {
    perusteet: null,
    opetussuunnitelmat: null,
    tiedotteet: null,
    koulutustyyppi: null,
    perusteId: null,
    async reload() {},
    constructor(koulutustyyppi: string,
      perusteId?: number) {
      this.koulutustyyppi = koulutustyyppi;
      this.perusteId = perusteId || null;
    },
    ...config,
  } as PerusteKoosteStore);
};


export const perusteStoreMock = (config: Partial<PerusteStore> = {}) => {
  return wrap(PerusteStore.prototype, {
    uusimmat: null,
    perusteet: null,
    async getUusimmat() {},
    async getYleisetPerusteet() {},
    ...config,
  } as PerusteStore);
};


export const tiedoteStoreMock = (config: Partial<TiedoteStore> = {}) => {
  return wrap(TiedoteStore.prototype, {
    uusimmatTiedotteet: null,
    async getUusimmat() {},
    ...config,
  } as TiedoteStore);
};


export const perusteDataStoreMock = (config: Partial<PerusteDataStore> = {}) => {
  return wrap(PerusteDataStore.prototype, {
    ...config,
  } as PerusteDataStore);
};
