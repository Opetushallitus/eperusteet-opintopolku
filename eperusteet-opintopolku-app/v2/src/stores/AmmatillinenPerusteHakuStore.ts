import { Store, Getter, State } from '@shared/stores/store';
import { PerusteenJulkaisuData } from '@shared/api/eperusteet';
import { Kielet } from '@shared/stores/kieli';
import { julkaistutPerusteet, JulkaistutPerusteetQuery } from '@/api/eperusteet';
import _ from 'lodash';
import { IPerusteHakuStore } from './IPerusteHakuStore';

@Store
export class AmmatillinenPerusteHakuStore implements IPerusteHakuStore {
  @State() public perusteet: PerusteenJulkaisuData[] | null = null;
  @State() public page = 0;
  @State() public pages = 0;
  @State() public total = 0;
  @State() public perPage = 10;

  @State()
  public filterdata: JulkaistutPerusteetQuery = {
    nimi: '',
    koulutustyyppi: [
      'koulutustyyppi_1',
      'koulutustyyppi_11',
      'koulutustyyppi_12',
      'koulutustyyppi_5',
      'koulutustyyppi_18',
    ],
    koulutusvienti: false,
    tuleva: true,
    siirtyma: false,
    voimassaolo: true,
    poistunut: false,
    sisaltotyyppi: 'kaikki',
  };

  constructor(data: JulkaistutPerusteetQuery = { }) {
    this.filterdata = {
      ...this.filterdata,
      ...data,
    };
  }

  @Getter((state) => {
    return [
      'tuleva',
      'voimassaolo',
      'siirtyma',
      'poistunut',
    ];
  })
  public readonly toggles!: string[];

  @Getter((state) => ({
    ...state.filterdata,
    sivu: state.page,
    sivukoko: state.perPage,
    kieli: Kielet.getSisaltoKieli.value,
  }))
  public readonly filters!: JulkaistutPerusteetQuery;

  public readonly fetch = _.debounce(async () => {
    this.perusteet = null;
    const result = await julkaistutPerusteet(this.filters);

    this.total = result['kokonaismäärä'];
    this.page = result.sivu;
    this.perPage = result.sivukoko;
    this.pages = result.sivuja;
    this.perusteet = result.data;
  }, 500);

  async updateFilters(filters: JulkaistutPerusteetQuery) {
    this.filterdata = {
      ...this.filters,
      ...filters,
    };
    await this.fetch();
  }
}
