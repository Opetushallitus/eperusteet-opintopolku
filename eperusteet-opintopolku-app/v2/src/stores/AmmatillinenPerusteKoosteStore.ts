import Vue from 'vue';
import VueCompositionApi, { reactive, computed, ref, watch } from '@vue/composition-api';
import _ from 'lodash';
import { JulkinenApi, OpetussuunnitelmaDto } from '@shared/api/amosaa';
import { PerusteDto, TiedoteDto, Perusteet } from '@shared/api/eperusteet';
import { Kielet } from '@shared/stores/kieli';
import { tiedoteQuery } from '@/api/eperusteet';

Vue.use(VueCompositionApi);

export class AmmatillinenPerusteKoosteStore {
  public state = reactive({
    peruste: null as PerusteDto | null,
    opetussuunnitelmat: null as OpetussuunnitelmaDto | null,
    tiedotteet: null as TiedoteDto | null,
  })

  constructor(private perusteId: number) {
    this.fetch();
  }

  public readonly peruste = computed(() => this.state.peruste);
  public readonly tiedotteet = computed(() => this.state.tiedotteet);
  public readonly opetussuunnitelmat = computed(() => this.state.opetussuunnitelmat);

  public async fetch() {
    this.state.peruste = (await Perusteet.getPerusteenTiedot(this.perusteId)).data;

    this.state.opetussuunnitelmat = ((await JulkinenApi.findOpetussuunnitelmat(
      this.state.peruste.diaarinumero,
      this.state.peruste.id,
      undefined,
      undefined,
      0,
      100,
      '',
      Kielet.getUiKieli.value,
    )).data as any).data;

    const vanhat = (await tiedoteQuery({
      sivukoko: 100,
      perusteId: this.perusteId,
    }));

    const uudet = (await tiedoteQuery({
      sivukoko: 100,
      perusteIds: [this.perusteId],
    }));

    this.state.tiedotteet = _.chain([
      ...vanhat,
      ...uudet,
    ])
      .uniqBy('id')
      .sortBy('luotu')
      .reverse()
      .value() as any;
  }
}
