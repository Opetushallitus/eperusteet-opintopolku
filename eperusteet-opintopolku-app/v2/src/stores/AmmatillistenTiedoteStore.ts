import Vue from 'vue';
import VueCompositionApi, { reactive, computed, ref, watch } from '@vue/composition-api';
import { TiedoteDto, Tiedotteet, PageTiedoteDto } from '@shared/api/eperusteet';
import _ from 'lodash';
import { ITiedotteetProvider } from '@shared/stores/types';
import { TiedoteQuery } from '@shared/api/types';

Vue.use(VueCompositionApi);

export class AmmatillistenTiedoteStore implements ITiedotteetProvider {
  private state = reactive({
    tiedotteet: null as TiedoteDto[] | null,
    query: {} as TiedoteQuery,
    options: null as TiedoteQuery | null,
    isLoading: true,
    kokonaismaara: null as number | null,
  });

  public readonly options = computed(() => this.state.tiedotteet);
  public readonly tiedotteet = computed(() => this.state.tiedotteet);
  public readonly isLoading = computed(() => this.state.isLoading);
  public readonly kokonaismaara = computed(() => this.state.kokonaismaara);
  public readonly perusteenTiedotteet = computed(() => []);
  public readonly query = computed(() => this.state.query);

  async init(options: TiedoteQuery) {
    this.state.options = options;
    this.state.query.koulutusTyyppi = options.koulutusTyyppi;
  }

  public async fetch() {
    this.state.isLoading = true;
    this.state.kokonaismaara = null;
    const res = (await Tiedotteet.findTiedotteetBy(
      0,
      9999,
      this.state.query.kieli,
      this.state.query.nimi,
      this.state.query.perusteId,
      this.state.query.perusteeton,
      this.state.query.julkinen,
      this.state.query.yleinen,
      this.state.query.tiedoteJulkaisuPaikka,
      this.state.query.perusteIds,
      this.state.query.koulutusTyyppi,
    )).data as any;
    this.state.tiedotteet = res.data;
    this.state.kokonaismaara = res.data.kokonaismaara;
    this.state.isLoading = false;
  }

  public async save() {
    // not supported
  }

  public async delete() {
    // not supported
  }
}
