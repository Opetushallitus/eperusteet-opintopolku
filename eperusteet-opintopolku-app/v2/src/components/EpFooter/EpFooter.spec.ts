import { mount, createLocalVue } from '@vue/test-utils';
import EpFooter from './EpFooter.vue';
import { Kielet } from '@shared/stores/kieli';
import { mocks, stubs } from '@shared/utils/jestutils';
import VueI18n from 'vue-i18n';
import { Kaannos } from '@shared/plugins/kaannos';


describe('EpFooter', () => {
  const localVue = createLocalVue();
  localVue.use(VueI18n);
  Kielet.install(localVue);
  localVue.use(new Kaannos());

  test('Renders', async () => {
    const wrapper = mount(EpFooter as any, {
      localVue,
      stubs: {
        ...stubs,
      },
      mocks: {
        ...mocks,
      },
    });

    await localVue.nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
