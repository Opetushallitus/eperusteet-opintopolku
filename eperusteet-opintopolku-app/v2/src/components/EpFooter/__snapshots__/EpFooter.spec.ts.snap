// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpFooter Renders 1`] = `
<footer>
  <div class="horizontal-line"><a href="https://www.oph.fi/fi"><img src="" alt="oph"></a></div>
  <div class="footer-content">
    <div class="row justify-content-around">
      <div class="col-lg-2 col-md-6 col-sm-12 mb-5"><a href="https://opintopolku.fi/wp/fi"><img src="" alt="opintopolku" class="opintopolku"></a></div>
      <div class="col-lg-2 col-md-6 col-sm-12 mb-5"><span class="font-weight-bold">eperusteet-virkailija-linkki</span>
        <div class="linkki"><a href="http://virkailija.opintopolku.fi" rel="noopener noreferrer" target="_blank"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="ulkoinen-linkki" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" class="mr-1 svg-inline--fa fa-ulkoinen-linkki fa-w-16 fa-fw">
              <path fill="currentColor" d="M9.333 0v2.667H2.667v14.666h14.666v-6.666H20v6.963A2.377 2.377 0 0117.63 20H2.37A2.37 2.37 0 010 17.63V2.37A2.37 2.37 0 012.37 0h6.963zM20 0v8h-2.667V4.552l-7.77 7.772-1.887-1.886 7.771-7.772H12V0h8z" class=""></path>
            </svg> <span>virkailija.opintopolku.fi</span></a></div>
      </div>
      <div class="col-lg-2 col-md-6 col-sm-12 mb-5"><span class="font-weight-bold">anna-palautetta</span>
        <div class="linkki"><a href="mailto:eperusteet@opintopolku.fi" rel="noopener noreferrer" target="_blank"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="envelope" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" class="mr-1 svg-inline--fa fa-envelope fa-w-16 fa-fw">
              <path fill="currentColor" d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z" class=""></path>
            </svg> <span>eperusteet@opintopolku.fi</span></a></div>
      </div>
      <div class="col-lg-2 col-md-6 col-sm-12 mb-5"><a href="https://opintopolku.fi/konfo/fi/sivu/eperusteet-saavutettavuusseloste">saavutettavuusseloste</a></div>
      <div class="col-lg-2 col-md-6 col-sm-12 mb-5"><a href="http://localhost:9020/vanha-ui/#/fi">vanha_kayttoliittyma</a></div>
    </div>
  </div>
</footer>
`;
